File: picker/src/main/java/ir/shahabazimi/instagrampicker/classes/InstaPickerSharedPreference.java
package ir.shahabazimi.instagrampicker.classes;

import android.content.Context;
import android.content.SharedPreferences;

public class InstaPickerSharedPreference {
    private static InstaPickerSharedPreference instance;
    private final SharedPreferences sp;

    private InstaPickerSharedPreference(Context context) {
        sp = context.getSharedPreferences("InstagramPickerSharedPreference", 0);
    }

    public static InstaPickerSharedPreference getInstance(Context context) {
        if (instance == null)
            instance = new InstaPickerSharedPreference(context);

        return instance;

    }

    public boolean getCameraPermission(){
        return sp.getBoolean("camera",false);

    }

    public void setCameraPermission(){
        SharedPreferences.Editor editor=sp.edit();
        editor.putBoolean("camera",true);
        editor.apply();
    }

    public boolean getStoragePermission(){
        return sp.getBoolean("external",false);

    }
    public void setStoragePermission(){
        SharedPreferences.Editor editor=sp.edit();
        editor.putBoolean("external",true);
        editor.apply();
    }
}

File: picker/src/main/java/ir/shahabazimi/instagrampicker/classes/InstaPickerSharedPreference.kt
package ir.shahabazimi.instagrampicker.classes

import android.content.Context
import androidx.core.content.edit
import androidx.security.crypto.EncryptedSharedPreferences
import androidx.security.crypto.MasterKey

class InstaPickerSharedPreference private constructor(ctx: Context) {
    private val masterKey =
        MasterKey.Builder(ctx).setKeyScheme(MasterKey.KeyScheme.AES256_GCM).build()
    private val sp = EncryptedSharedPreferences.create(
        ctx,
        "InstagramPickerSharedPreference",
        masterKey,
        EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
        EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
    )


    companion object {
        @Volatile
        private var instance: InstaPickerSharedPreference? = null
        @JvmStatic
        operator fun invoke(ctx: Context): InstaPickerSharedPreference =
            instance ?: synchronized(this) {
                instance ?: InstaPickerSharedPreference(ctx).also { instance = it }
            }
    }

    fun getCameraPermission() = sp.getBoolean("camera", false)
    fun setCameraPermission() = sp.edit { putBoolean("camera", true) }

    fun getStoragePermission() = sp.getBoolean("external", false)
    fun setStoragePermission() = sp.edit { putBoolean("external", true) }
}

